docs/articles/documentation/test-api/selecting-page-elements/selectors.md

```diff
@@ -123,221 +131,311 @@ You can initialize a selector with any of these objects.
-use the `select` method of the [test controller](../test-code-structure.md#test-controller).
+### Get Selector Matching Set Length
 
-```text
-t.select( init [, options] )
+Functions and CSS selector strings that initialize a selector may return
+a single matching DOM element on the page, multiple elements or nothing.
+You can use the following Selector properties to check whether the matching
+elements exist or get a number of them.
+
+Property | Type | Description
+------ | ----- | -----
+`exists` | Boolean | `true` if at least one matching element exists.
+`count` | Number | The number of matching elements.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `Example page`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const osCount            = Selector('.column.col-2 label').count;
+    const submitButtonExists = Selector('#submit-button').exists;
+
+    await t
+        .expect(osCount).eql(3)
+        .expect(submitButtonExists).ok();
+});
 ```
 
-Parameter              | Type     | Description
----------------------- | -------- | ------------------------------------------------------------------
-`init`                 | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies a DOM node to be selected. See [Selector Initializers](#selector-initializers).
-`options`&#160;*(optional)* | Object   | See [Options](selector-options.md).
+Note that selector property getters are asynchronous.
 
-The following example shows how to get a DOM element by ID with `t.select`.
+### Obtain Element State
+
+Selectors and promises returned by selectors expose API to get state (size, position, classes, etc.) of matched element.
+See [DOM Node State](./dom-node-state.md). Note that these methods and property getters
+are asynchronous so you can obtain element's property from a browser in the following way:
 
 ```js
+const headerText = await Selector('#header').textConent;
+```
+
+For example:
+
+```js
+
+import { Selector } from 'testcafe';
+
 fixture `My fixture`
-    .page `http://www.example.com/`;
+    .page('http://devexpress.github.io/testcafe/example/');
 
-test('My Test', async t => {
-    const header = await t.select(() => document.getElementById('header'));
+const windowsInput = Selector('#windows');
+
+test('Obtain Element State', async t => {
+    await t.click(windowsInput);
+
+    const windowsInputChecked = await windowsInput.checked; // returns true
 });
-```
 
-## Using Selectors to Define Action Targets
+ ```
 
-You can pass selectors to [test actions](../actions/index.md) to use the returned element as the action target.
+#### DOM Node Snapshot
+
+If you need to get all state properties of the DOM element at once call the selector
+with the `await` keyword like you would do with regular async functions.
+It returns a *DOM Node Snapshot* that contains information about the node's size, position, classes, parent and child nodes, etc.
```

Comment:
Don't enumerate properties, just note that snapshot contains all property values exposed by selector in single object.